### PR TITLE
DYN-6470 updates

### PR DESCRIPTION
**Pull request by saintentropy**
### Purpose

This PR reverts some the changes in https://github.com/DynamoDS/Dynamo/pull/14658.  Additional scenarios were found where depending on the initialization of the DynamoModel sync issues could still arise.  

Specially:

1. API user access `PreferenceSettings` via `Instance` and makes a modification to the runtime object
2. `DynamoModel` constructor is called with no `PreferenceSettings` set
3. `DynamoModel` constructor creates a new `PreferenceSettings` and `Instance` is set to the new object.

Now the modifications applied in step 1 are lost.

This PR resets the primary driver of `PreferenceSettings` to `DynamoModel.PreferenceSettings.`.  `Instance` will provide a `PreferenceSetting` object based on the following order of criteria:

1. if `DynamoMode` l present, the `DynamoModel.PreferenceSettings` object is returned,
2. else, if a valid setting xml file exists, the `PreferenceSettings` object de-serialized from the xml file is returned,
3. else, if no `DynamoModel` and no valid xml file exists, a new `PreferenceSettings` object returned

`Instance` is also moved to an internal method until we can improve the delivery and synchronization guarantees.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Instance API is now internal

### Reviewers

@mjkkirschner @pinzart90 @aparajit-pratap 

### FYIs

@QilongTang

### CODE REVIEW

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -753,7 +753,8 @@ protected DynamoModel(IStartConfiguration config)
                 if (migrator != null)
                 {
                     var isFirstRun = PreferenceSettings.IsFirstRun;
-                    PreferenceSettings.Instance = migrator.PreferenceSettings;
+                    PreferenceSettings = migrator.PreferenceSettings;
```
Comment (aparajit-pratap) [inline src/DynamoCore/Models/DynamoModel.cs:756]: Shouldn't `PreferenceSettings.Instance` be updated here as well?

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 2]
```diff
@@ -48,12 +48,21 @@ internal static void CopyProperties(this PreferenceSettings source, PreferenceSe
...
+        /// 1) if DynamoModel present, the DynamoModel.PreferenceSettings object is returned,
+        /// 2) else, if a valid setting xml file exists, the PreferenceSettings object de-serialized from the xml file is returned,
+        /// 3) else, if no DynamoModel and no valid xml file exists, a new PreferenceSettings object returned
+        /// Note that Instance is a runtime object only.  No changes to the PreferenceSettings will be persisted on disk with condition 2 or 3.
+        /// User of Instance must initiate save operations to insure persistence of modifications to the PreferenceSettings model.
+        /// In some cases even the save will not guarantee persistence of modifications depending on the startup of DynamoModel.
+        /// </summary>
         [XmlIgnore]
-        public static PreferenceSettings Instance { get; internal set; } = lazy.Value;
+        internal static PreferenceSettings Instance { get; set; } = lazy.Value;
```
Comment (aparajit-pratap) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:65]: The whole singleton implementation for this class is flawed. Typically in a singleton pattern, the instance should be public and the constructors should be all private.

Comment (aparajit-pratap) [reply src/DynamoCore/Configuration/PreferenceSettings.cs:65]: Unfortunately, this class has a public constructor.

File: src/DynamoCore/Models/DynamoModel.cs (hunk 1 above)
Comment (saintentropy) [reply src/DynamoCore/Models/DynamoModel.cs:756]: I thought originally and had it that way.  @QilongTang pointed out that it shouldn't need an update.  

We have:
PreferenceSettings = foo;
Instance = PreferenceSettings
PreferenceSettings = otherFoo;

I don't think you need to reset the Instance right?

Comment (pinzart90) [reply src/DynamoCore/Models/DynamoModel.cs:756]:
```
        string PreferenceSettings = "val1";
        string Instance = PreferenceSettings;
        PreferenceSettings = "val2";
        
        Console.WriteLine (PreferenceSettings);
        Console.WriteLine (Instance);
       -----------
        val2
        val1
        ----------
```

Comment (saintentropy) [reply src/DynamoCore/Models/DynamoModel.cs:756]: Ok let me fix that :-)

**Review by pinzart90 — APPROVED** [review]

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**